### PR TITLE
Override errexit in kube-up for validate-cluster

### PR DESCRIPTION
**Pull request by gmarek**

### CODE REVIEW

File: cluster/kube-up.sh [hunk 1]
```diff
@@ -53,19 +53,20 @@ echo "... calling kube-up" >&2
 kube-up
 
 echo "... calling validate-cluster" >&2
-if [[ "${EXIT_ON_WEAK_ERROR}" == "true" ]]; then
-	validate-cluster
+# Override errexit
+if validate-cluster; then
```
Comment (spxtr) [inline cluster/kube-up.sh]: You might be able to get away with `(validate-cluster) && validate_result="$?" || validate_result="$?"`.

Comment (gmarek) [inline cluster/kube-up.sh]: I'll check if this works. Thanks.

**Comment by gmarek** [comment]
Done. @zmerlynn/@spxtr PTAL

File: cluster/kube-up.sh [hunk 2]
```diff
@@ -53,19 +53,16 @@ echo "... calling kube-up" >&2
...
-	if [[ ${validate_result} != "0" ]]; then
-		if [[ "${validate_result}" == "1" ]]; then
-			exit 1
-		elif [[ "${validate_result}" == "2" ]]; then
-			echo "...ignoring non-fatal errors in validate-cluster" >&2
-		else
-			echo "Got unknown validate result: ${validate_result}"
-		fi
+# Override errexit
+(validate-cluster) && validate_result="$?" || validate_result="?"
```
Comment (spxtr) [inline cluster/kube-up.sh]: Should read `$?`, not `?`.

Comment (gmarek) [inline cluster/kube-up.sh]: Good catch. Thanks! PTAL

**Comment by spxtr** [comment]
In this 76-line file we use two spaces, four spaces, and tabs. That's great!

Looks fine to me but I don't know what it's doing. A comment might be appreciated.

**Comment by gmarek** [comment]
Added a comment. PTAL.

**Comment by gmarek** [comment]
@zmerlynn @spxtr PTAL

**Comment by zmerlynn** [comment]
@k8s-bot e2e test this issue: #21995 (closed, but should be fixed now?)

**Pull request merged**